### PR TITLE
Add ID parsing and validation functions for `azurerm_management_group`

### PR DESCRIPTION
**Pull request by ArcturusZhang**
* Add ID parsing function for `azurerm_management_group`
* Add ID validation function for `azurerm_management_group`
* Add validation function for management group name (`group_id`)

### CODE REVIEW

File: azurerm/internal/services/managementgroup/parse/management_group.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	"fmt"
+	"regexp"
+)
+
+type ManagementGroupId struct {
+	GroupID string
+}
+
+func ManagementGroupID(input string) (*ManagementGroupId, error) {
+	regex := regexp.MustCompile(`^/providers/[Mm]icrosoft\.[Mm]anagement/management[Gg]roups/`)
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/managementgroup/parse/management_group.go]: Well, for other resource ids, we usually just ignore the value for `providers`, this is the reason that I ignore the cases in the `providers` segment.
And as for the `managementGroups` segment, there are some resources are not keep the cases for management group IDs (for instance, the policy insight remediation). For flexibility and simplification, I just ignored the cases in this segment, similar to the same logic for `resourceGroups` in `azure.ParseAzureResourceID`

File: azurerm/internal/services/managementgroup/parse/management_group.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	// Split the input ID by the regex
+	segments := regex.Split(input, -1)
+	if len(segments) != 2 {
+		return nil, fmt.Errorf("Unable to parse Management Group ID %q: expected id to have two segments after splitting", input)
+	}
+	groupID := segments[1]
+	_, errs := ValidateManagementGroupName(groupID, "")
+	if len(errs) != 0 {
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/managementgroup/parse/management_group.go]: Done

**Comment by ArcturusZhang** [comment]
Hi @katbyte I have resolved some of the comments.
In the meantime, I have a question that is it common that we validate the name of the resource when parsing it ID? It is quite weird that the name validation function has to be in the `parse` directory due to the cyclic import issue if it is in the `validate` directory.

**Comment by katbyte** [comment]
@ArcturusZhang, typically no, we don't bother validating the names because we expect anything from azure to be correct, and any we generate ourselves should be validate by the schema

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang, typically no, we don't bother validating the names because we expect anything from azure to be correct, and any we generate ourselves should be validate by the schema

Well, this concern comes up with this issue: https://github.com/terraform-providers/terraform-provider-azurerm/issues/5677
User uses an ID with a extra segment in this issue, like `/providers/Microsoft.Management/managementGroups/foo/bar`.
Currently I am using the name validation to validate if there is extra segments.
Please let me tweak this a little so that the ID parsing function would not depend on the name validation logic.

**Comment by ArcturusZhang** [comment]
Hi @katbyte I have made some changes regarding your comments, please have a look, thanks!

**Comment by katbyte** [comment]
@ArcturusZhang, as you have pulled these changes into #5746 i am going to close this PR

**Pull request closed without merging**